### PR TITLE
Reorganize angular project

### PR DESCRIPTION
**Pull request by gnomeontherun**
This is done in 4 commits for a reason. The first is to generate the new project, the second moves the files around (so to keep track of git history more easily), and the last is a set of changes made to make the project build and test.

## This PR

* Creates a new Angular CLI project
* Moves the existing clr-angular library
* Updates build and configuration
* Updates prettier and configuration

## Next Steps

* Dev and website apps are both broken, and intended to be removed separately
* Remove gemini, since dev app is not functional it won't help anymore in master
* Replace npm with yarn to use workspaces

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [ ] No

### CODE REVIEW

File: packages/angular/angular.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
   "version": 1,
   "cli": {
-    "packageManager": "yarn"
+    "packageManager": "npm"
```
Comment (coryrylan) [inline packages/angular/angular.json:5]: Should this be yarn not npm?

File: packages/angular/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,17131 @@
+{
+  "name": "@clr/angular",
```
Comment (coryrylan) [inline packages/angular/package-lock.json:2]: Is this lockfile generated by yarn?

File: package.json [hunk 3]
```diff
@@ -31,17 +31,10 @@
...
-    "preangular:lint": "npm link ./scripts/eslint-plugin-clarity",
-    "angular:lint": "eslint -c src/clr-angular/.eslintrc.js \"src/clr-angular/**/*.ts\" \"src/dev/**/*.ts\" \"src/website/**/*.ts\" && stylelint \"src/clr-angular/**/*.scss\" \"src/dev/**/*.scss\" \"src/website/**/*.scss\"",
-    "angular:lint:fix": "eslint --fix -c src/clr-angular/.eslintrc.js \"src/clr-angular/**/*.ts\" \"src/dev/**/*.ts\" \"src/website/**/*.ts\" && stylelint --fix \"src/clr-angular/**/*.scss\" \"src/dev/**/*.scss\" \"src/website/**/*.scss\"",
-    "angular:golden": "ts-api-guardian --out dist/golden/clr-angular.d.ts --stripExportPattern '^Çlr' dist/clr-angular/index.d.ts",
-    "angular:golden:fix": "npm run angular:build && ts-api-guardian --out golden/clr-angular.d.ts --stripExportPattern '^Çlr' dist/clr-angular/index.d.ts",
-    "angular:golden:test": "diff -u golden/clr-angular.d.ts dist/golden/clr-angular.d.ts",
+    "angular:build": "yarn --cwd packages/angular build",
+    "angular:test": "yarn --cwd packages/angular test",
+    "angular:test:watch": "yarn --cwd packages/angular test:watch",
+    "angular:lint": "yarn --cwd packages/angular lint",
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply package.json:37]: I switched back to npm and have another commit to push for these. I started with yarn, but mixing npm and yarn in the same project was causing issues when running the larger set of commands. Yarn can be a separate change.

File: packages/angular/package.json [hunk 4]
```diff
@@ -29,18 +38,34 @@
...
-    "@types/jasmine": "~3.5.0",
+    "@types/jasmine": "3.4.6",
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "^12.11.1",
+    "@typescript-eslint/eslint-plugin": "2.30.0",
+    "@typescript-eslint/parser": "2.30.0",
     "codelyzer": "^5.1.2",
-    "jasmine-core": "~3.5.0",
+    "cpy-cli": "3.1.0",
+    "eslint": "6.8.0",
```
Comment (hippee-lee) [inline packages/angular/package.json:48]: woohoo!

File: packages/angular/projects/clr-angular/src/accordion/stepper/stepper-panel.ts [hunk 5]
```diff
@@ -56,7 +56,7 @@ export class ClrStepperPanel extends ClrAccordionPanel implements OnInit {
   }
 
   constructor(
-    @Inject(PLATFORM_ID) private platformId: Record<string, any>,
+    @Inject(PLATFORM_ID) private platformId: any,
```
Comment (hippee-lee) [inline packages/angular/projects/clr-angular/src/accordion/stepper/stepper-panel.ts:59]: Was the Record typing causing an issue?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -12,7 +12,7 @@ jobs:
       - uses: actions/setup-node@v1
         with:
           node-version: '12.x'
-      - run: npm install && npm --prefix packages/icons install
+      - run: npm install && npm --prefix packages/icons install && yarn --cwd packages/angular
```
Comment (hippee-lee) [inline .github/workflows/build.yml]: Yarn gets removed in the commit after `get angular building and testing` Correct?

Comment (gnomeontherun) [reply .github/workflows/build.yml]: Yes, I missed this reference before my first push.

File: packages/angular/projects/clr-angular/src/accordion/stepper/stepper-panel.ts (hunk 5 above)
Comment (gnomeontherun) [reply packages/angular/projects/clr-angular/src/accordion/stepper/stepper-panel.ts:59]: The stricter Angular compiler setting was unable to have a generic in the typing for something explicitly using `@Inject()`. There is a bug for it, but would have to dig it up.

File: .travis.yml [hunk 7]
```diff
@@ -1,35 +0,0 @@
-language: node_js
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply .travis.yml:1]: It should but the check is still running on this PR but there is no Travis job. Need to figure that one out.

File: packages/angular/browserslist [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/browserslist/browserslist#queries
+
+# You can see what browsers were selected by your queries by running:
+#   npx browserslist
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+not IE 9-11 # For IE 9-11 support, remove 'not'.
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply packages/angular/browserslist:12]: I should make sure it says `IE11`, as that is what we have in the root `browserslist`

**Pull request merged**